packages/relay-signer/src/relayer.ts

```
@@ -59,7 +59,12 @@ export class Relayer implements IRelayer {
 
   public constructor(credentials: RelayerParams) {
     this.credentials = credentials;
-    if (isActionCredentials(credentials)) {
+    const defenderEnv = process.env.DEFENDER_ENV;
+    const errorMessage = `Missing credentials for creating a Relayer instance. If you are running this code in an Action, make sure you pass the "credentials" parameter from the handler to the Relayer constructor. If you are running this on your own process, then pass an object with the "apiKey" and "apiSecret" generated by the relayer.`;
+    if (defenderEnv === 'action') {
+      if (!isActionCredentials(credentials)) {
```

Some users are using multiple relayers in Action. For that they pass api key and secret.

Would that continue to work with these changes or error would be thrown?

---

@MCarlomagno can you help me with this? I am not totally sure

---

Great catch @zeljkoX , it was my bad, I suggested doing this change to relayers too, and I realized this is only necessary in kvstore (and actually just throwing a proper error works). When I created the ticket bram was trying to use the same Defender client to instantiate the kv store AND the relayer, and the error thrown was confusing.

The idea is that we want to throw an error when kvstore is instantiated in an action using other credentials than `isActionCreateParams`, so these changes are not needed here